### ISSUE TITLE
too much sensitive information in api call "instance"

### ISSUE THREAD

**Issue by Cyborgscode** [friendica/friendica#8166]
I just found out, that Friendica 2019.12 is leaking sensitive Information about the admin account to anyone asking. Revealing the admin emailaddress in an insecure web call, opens the way for :

- Spam / Trojans / Phising
- Passwordreset attacks
etc.

For what reason does the api call tell the caller, that the account is "locked" or not? 

IF anything about the userbase is revealed, it should be a hashid to "add" a contact, but thats it. If the contact accepted the friendship request, the behaviour could change, but not to "any facebookcrwaler bot asking" .

Real World Example:

domainnames replaced by "hostname" 
Adminemail replaced by "full.featured@email.address"
mainusername replaced by "Username"
```
$ curl https://hostname/api/v1/instance
{"uri":"https:\/\/hostname","title":"Site Name","description":"","email":"**full.featured@email.address**","version":"2019.12","urls":[],"stats":{"user_count":null,"status_count":null,"domain_count":null},"thumbnail":"https:\/\/hostname","languages":["de"],"max_toot_chars":200000,"registrations":false,"approval_required":false,"contact_account":{"id":1,"username":"Username","acct":"Username","display_name":"Username","locked":false,"created_at":"2019-03-18T21:47:32Z","followers_count":7,"following_count":5,"statuses_count":44,"note":"My complete describition is hidden in the dark...","url":"https:\/\/hostname\/profile\/Username","avatar":"https:\/\/hostname\/photo\/profile\/1.jpg","avatar_static":"https:\/\/hostname\/photo\/profile\/1.jpg","header":"","header_static":"","emojis":[],"moved":null,"fields":null,"bot":false}}
```

### Expected behavior

not revealing private and sensitive informations.

### Friendica version you encountered the problem

2019.12 , but i'm afraid "any" version will leak infos.

**Comment by MrPetovan** [comment]
An email address isn't meant as a secret information. It is meant as a point of contact, and as such is important in a Friendica node public information block. Similarly, the locked flag is a public one.

**Comment by annando** [comment]
We could use the mail address that is used for sending mails.

**Comment by SpcCw** [comment]
> We could use the mail address that is used for sending mails.

Would be much better. Exposing sign-in email of server administrator is certainly not a good practice.

**Comment by MrPetovan** [comment]
> Exposing sign-in email of server administrator is certainly not a good practice.

This is a completely moot point, user's nicknames are exposed all the time and are used for login. This is the point of password and two-factor authentication, because nobody should expect their login (nickname/email) to be secret.

**Comment by nupplaphil** [comment]
The difference is that with a username, you can "only" try to bruteforce the account.
With an email, you can try every kind of email-phising possibility we know. It delegates the security-problem from one point (Friendica) to two points (Friendica & mail-provider).

**Comment by MrPetovan** [comment]
Yes but the email phishing problem isn't specific to Friendica, it's a problem with email in general. Whether we provide a login email or a non-login contact email in this API call, the problem will stay the same.

I don't mind the suggested change, note, I'm just saying it has nothing to do with either privacy or security with regard to Friendica. If you are susceptible to phishing/spam on your email addresses, Friendica won't prevent it from happening, even if we cleared out this API call of any email address.

**Comment by SpcCw** [comment]
> Yes but the email phishing problem isn't specific to Friendica, it's a problem with email in general.

Yes, it is not Friendica specific and applies to many services. This is why most services don't show (at least by default) administrative (or even user) e-mail addresses unless it is required for operation :)

It is just added attack surface, not critical at all but still not really needed.

E.g. people might use for their admin account their private address they wouldn't want to be public. I don't recall if there are any warnings in the documentation about admin address becoming publicly visible (but maybe I just missed it?)

**Comment by MrPetovan** [comment]
You didn’t miss any warning, which is one of the reasons why I support this change.

**Comment by nupplaphil** [comment]
closed per #15420
